client/blocks/comments/post-comment-list.jsx

```diff
@@ -55,26 +71,81 @@ class PostCommentList extends React.Component {
 		activeEditCommentId: null,
 	};
 
-	componentWillMount() {
-		const { post: { ID: postId, site_ID: siteId } } = this.props;
+	/**
+	 * Should we scroll down to a comment? Only if we have satisfied these conditions:
+	 * 1. there is a startingCommentId
+	 * 2. the comment has loaded and is on the DOM
+	 * 3. we haven't already scrolled to it yet
+	 * 4. we have loaded some comments above + below it already (or there is only 1 comment)
+	 *
+	 * @param {object} props - the propes to use when evaluating if window should be scrolled down to a comment.
+	 * @returns {boolean} - whether or not we should scroll to a comment
+	 */
+	shouldScrollToComment = ( props = this.props ) =>
+		props.startingCommentId &&
+		props.commentsTree[ this.props.startingCommentId ] &&
+		props.commentsFetchingStatus.hasReceivedBefore &&
+		props.commentsFetchingStatus.hasReceivedAfter &&
+		! this.hasScrolledToComment &&
+		window.document.getElementsByName( `comment-${ props.startingCommentId }` ).length > 0;
+
+	shouldFetchInitialComment = ( { startingCommentId, commentsTree } ) =>
+		startingCommentId && commentsTree && ! commentsTree[ startingCommentId ];
+
+	shouldFetchInitialPages = ( { commentsTree, startingCommentId } ) =>
+		startingCommentId &&
+		commentsTree[ startingCommentId ] &&
+		this.props.commentsTree[ startingCommentId ] &&
+		! this.alreadyLoadedInitialSet;
 
-		this.props.requestPostComments( siteId, postId, this.props.commentsFilter );
+	componentWillMount() {
+		const { post: { ID: postId, site_ID: siteId }, commentsFilter: status } = this.props;
+
+		if ( this.shouldFetchInitialComment( this.props ) ) {
+			this.props.requestComment( { siteId, commentId: this.props.startingCommentId } );
+		} else if ( this.shouldFetchInitialPages( this.props ) ) {
+			this.viewEarlierCommentsHandler();
+			this.viewLaterCommentsHandler();
+			this.alreadyLoadedInitialSet = true;
+		} else {
+			this.props.requestPostComments( { siteId, postId, status } );
+		}
 	}
 
 	componentWillReceiveProps( nextProps ) {
 		const nextSiteId = get( nextProps, 'post.site_ID' );
 		const nextPostId = get( nextProps, 'post.ID' );
 		const nextCommentsFilter = get( nextProps, 'commentsFilter' );
-
-		if (
-			nextSiteId &&
-			nextPostId &&
-			nextCommentsFilter &&
-			( this.props.post.site_ID !== nextSiteId ||
-				this.props.post.ID !== nextPostId ||
-				this.props.commentsFilter !== nextCommentsFilter )
+		const nextInitialComment = nextProps.initialComment;
+
+		if ( this.shouldFetchInitialComment( nextProps ) && ! nextProps.initialComment ) {
+			this.props.requestComment( { siteId: nextSiteId, commentId: nextProps.startingCommentId } );
+			this.hasScrolledToComment = false;
+		} else if ( this.shouldFetchInitialPages( nextProps ) ) {
+			this.viewEarlierCommentsHandler();
```

Comment:
I don't think so.  all of the comments are in redux and we use the oldest/newest comment in order to page.